cmd/mysterium_server/command_run/identity_handler_test.go

```diff
@@ -46,24 +63,58 @@ func Test_identityHandler_UseLast(t *testing.T) {
 
 	handler := NewNodeIdentityHandler(identityManager, client, cache, fakeSignerFactory)
 
-	id, err := handler.UseLast()
+	id, err := handler.UseLast("pass")
 	assert.Equal(t, fakeIdentity, id)
 	assert.Nil(t, err)
 
 	assert.Equal(t, "", client.RegisteredIdentity.Address)
+
+	assert.Equal(t, "abc", identityManager.LastUnlockAddress)
+	assert.Equal(t, "pass", identityManager.LastUnlockPassphrase)
+}
+
+func Test_identityHandler_UseLastUnlockFails(t *testing.T) {
+	identityManager.CleanStatus()
+	identityManager.UnlockFails = true
+	client := server.NewClientFake()
+	cache := identity.NewIdentityCacheFake()
+
+	fakeIdentity := identity.FromAddress("abc")
+	cache.StoreIdentity(fakeIdentity)
+
+	handler := NewNodeIdentityHandler(identityManager, client, cache, fakeSignerFactory)
+
+	_, err := handler.UseLast("pass")
+	assert.NotNil(t, err)
 }
 
 func Test_identityHandler_UseNew(t *testing.T) {
+	identityManager.CleanStatus()
 	client := server.NewClientFake()
 	cache := identity.NewIdentityCacheFake()
 
 	handler := NewNodeIdentityHandler(identityManager, client, cache, fakeSignerFactory)
 
-	id, err := handler.UseNew()
+	id, err := handler.UseNew("pass")
 	assert.Equal(t, newIdentity, id)
 	assert.Nil(t, err)
 
 	assert.Equal(t, newIdentity, client.RegisteredIdentity)
+
+	assert.Equal(t, newIdentity.Address, identityManager.LastUnlockAddress)
+	assert.Equal(t, "pass", identityManager.LastUnlockPassphrase)
+}
+
+func Test_identityHandler_UseNewUnlockFails(t *testing.T) {
+	identityManager.CleanStatus()
```

Comment:
Avoid having global state by constructing `identityManager` in each test

Reply:
Done